src/Stratis.Bitcoin.IntegrationTests/NodeBuilder.cs

```diff
@@ -328,55 +328,50 @@ public override string ToString()
 
     public class NodeBuilder : IDisposable
     {
-        public static NodeBuilder Create([CallerMemberNameAttribute] string caller = null, string version = "0.13.1")
+        /// <summary>
+        /// Deletes test folders. Stops "bitcoind" if required.
+        /// </summary>
+        /// <param name="folder">The folder to remove.</param>
+        /// <returns>Returns true if the folder was successfully removed and false otherwise.</returns>
+        public static bool CleanupTestFolder(string folder)
         {
-            version = version ?? "0.13.1";
-            if (!Directory.Exists("TestData"))
-                Directory.CreateDirectory("TestData");
-            var path = EnsureDownloaded(version);
-            caller = Path.Combine("TestData", caller);
-            bool retryDelete = true;
-            try
-            {
-                Directory.Delete(caller, true);
-                retryDelete = false;
-            }
-            catch (DirectoryNotFoundException)
-            {
-                retryDelete = false;
-            }
-            catch (UnauthorizedAccessException)
-            {
-            }
-            catch (IOException)
-            {
-            }
-            if (retryDelete)
+            bool tryKill = true;
+            while (true)
             {
+                try
+                {
+                    Directory.Delete(folder, true);
+                    return true;
+                }
+                catch (DirectoryNotFoundException)
+                {
+                    return true;
+                }
+                catch (Exception)
+                {
+                }
+                if (!tryKill)
+                    return false;
+                tryKill = false;
                 foreach (var bitcoind in Process.GetProcessesByName("bitcoind"))
                 {
                     if (bitcoind.MainModule.FileName.Contains("Stratis.Bitcoin.IntegrationTests"))
                     {
                         bitcoind.Kill();
                     }
                 }
-                // kill all dotnet instances no us
-                //foreach (var dotnet in Process.GetProcessesByName("dotnet"))
-                //{
-                //	if (Process.GetCurrentProcess().Id == dotnet.Id)
-                //		continue;
-                //	try
-                //	{
-                //		dotnet.Kill();
-                //	}
-                //	catch
-                //	{
-                //		// ignored
-                //	}
-                //}
                 Thread.Sleep(1000);
-                Directory.Delete(caller, true);
             }
+        }
+
+        public static NodeBuilder Create([CallerMemberNameAttribute] string caller = null, string version = "0.13.1")
+        {
+            version = version ?? "0.13.1";
+            if (!Directory.Exists("TestData"))
```

Comment:
no need to check `if (!Directory.Exists("TestData"))` as `Directory.CreateDirectory` will create the directory only if they exist.

Reply:
Same here.